src/libraries/System.Net.Mail/src/System/Net/Mail/SmtpClient.cs

```diff
@@ -606,7 +336,7 @@ public void SendAsync(MailMessage message, object? userToken)
                 throw new ObjectDisposedException(GetType().FullName);
             }
 
-            if (NetEventSource.IsEnabled) NetEventSource.Enter(this, message, userToken, _transport);
```

Comment:
most of this PR is just moving code but this piece isn't so I want to call it out: I didn't see a great way to split this method up given we only need `_transport` here for the network case.

I don't think we need to pass `_transport` to NetEventSource given it's already captured by the `NetEventSource.Associate(this, _transport);` in the `Initialize()` method so I removed it, however I'd appreciate a second look from someone more familiar with this 😄